### PR TITLE
improve github actions

### PR DESCRIPTION
**Pull request by SchSeba**
add cancel on progress and also run the virtual cluster tests only if all the light lanes like build and unit tests pass to save resources

### CODE REVIEW

**Comment by bn222** [comment]
Can you change the commit message to:

```
Speed up GitHub action runs and save resources

- Require short running actions (golangci, test, build) before starting slower virtual-k8s-cluster and only run virtual-ocp when virtual-k8s-cluster succeed
- Allow cancellation

These two changes together reduce the amount of resources used

```

**Comment by bn222** [comment]
LGTM

**Comment by SchSeba** [comment]
/hold 
this doesn't run the virtual cluster now ;(

**Comment by SchSeba** [comment]
I moved the virtual cluster to the same file as the build and unit test so we can have a dependency.
I also changed the name from Go to Test SR-IOV Operator to better represent the flow

**Review by zeeke — APPROVED** [review]
LGTM

File: hack/run-e2e-conformance-virtual-cluster.sh [hunk 1]
```diff
@@ -130,18 +130,67 @@ kcli ssh $cluster_name-ctlplane-0 << EOF
 sudo su
 echo '$insecure_registry' > /etc/containers/registries.conf.d/003-internal.conf
 systemctl restart crio
+
+echo '[connection]
```
Comment (adrianchiris) [inline hack/run-e2e-conformance-virtual-cluster.sh:134]: nit: define function and call it for each node ?

**Review by e0ne — APPROVED** [review]

File: hack/run-e2e-conformance-virtual-cluster.sh (hunk 1 above)
Comment (SchSeba) [reply hack/run-e2e-conformance-virtual-cluster.sh:136]: done

**Pull request merged**